packages/atomic/src/components/common/facets/facet-common.tsx

```diff
@@ -253,6 +263,101 @@ export function facetShouldBeInitiallyCollapsed(
   return facetIndex + 1 > collapseFacetsAfter;
 }
 
+interface VisibilitySortedFacets {
+  visibleFacets: BaseFacetElement[];
+  invisibleFacets: BaseFacetElement[];
+}
+export function sortFacetVisibility(
+  facetElements: BaseFacetElement[],
+  facetInfoMap: FacetInfoMap
+): VisibilitySortedFacets {
+  const visibleFacets: BaseFacetElement[] = [];
+  const invisibleFacets: BaseFacetElement[] = [];
+
+  facetElements.forEach((facet) => {
+    if (facetInfoMap[facet.facetId] && facetInfoMap[facet.facetId].isHidden()) {
+      invisibleFacets.push(facet);
+    } else {
+      visibleFacets.push(facet);
+    }
+  });
+
+  return {visibleFacets, invisibleFacets};
+}
+
+export function updateCollapsedState(
+  facets: BaseFacetElement[],
+  collapseFacetsAfter: number
+) {
+  if (collapseFacetsAfter === -1) {
+    return;
+  }
+
+  facets.forEach((facet, index) => {
+    facet.isCollapsed = facetShouldBeInitiallyCollapsed(
+      index,
+      collapseFacetsAfter
+    );
+  });
+}
+export function updateCollapseFacetsAfter(
+  generator: AutomaticFacetGeneratorElement,
+  index: number,
+  collapseFacetsAfter: number
+) {
+  if (!generator) {
+    return;
+  }
```

Comment:
The type of `generator` is non-optional and non-nullable, and it isn't scalar either, which implies that this condition would always be falsy and this `if` block wouldn't do anything. I recommend adjusting either the code outside of this function or making `generator` nullable.